views/actions/packages.js

```diff
@@ -65,26 +80,32 @@ export function getSubscribedPackages() {
   };
 }
 
-function createAddPackageRequest(trackingNumber) {
+function createAddPackageRequest(carrier, trackingNumber) {
   return {
     path: '/packages',
     method: 'POST',
     body: {
       package: {
+        carrier,
         trackingNumber,
       },
     },
   };
 }
 
-export function requestToAddPackage(trackingNumber) {
-  const request = createAddPackageRequest(trackingNumber);
+export function requestToAddPackage(carrier, trackingNumber) {
+  const request = createAddPackageRequest(carrier, trackingNumber);
   return (dispatch, getState) => {
+    query(request).then((response) => {
+      response.json().then((json) => {
+        if (!response.ok) {
+          dispatch(addAddPackageError());
+        } else {
+          dispatch(addPackage(json.id, json.carrier, json.trackingNumber));
+        }
+      });
+    }).catch((error) => {});
     dispatch(processAddPackage());
```

Comment:
This should be before query at line 99. Process is to help the ui identify that it is about to go do a long api request, so that we can load a spinner or loading screen etc.